### ISSUE TITLE
Build error occurred on macOS 10.14 Mojave

### ISSUE THREAD

**Issue by COx2** [hotwatermorning/Terra#17]
On macOS 10.14 Mojave, the source code below causes a build error.

https://github.com/hotwatermorning/Terra/blob/502b09adc7f901758c51ddc727061e54b507783a/Terra/prefix.hpp#L31

```
#if __has_include(<variant>)
    #include <variant>
#else
    #include <mpark/variant.hpp>
    namespace std {
        template<class... Args>
        using variant = mpark::variant<Args...>;
        using monostate = mpark::monostate;
        using mpark::get;
        using mpark::get_if;
    }
#endif
```
↓ for macOS 10.14 Mojave
```
//#if __has_include(<variant>)
//    #include <variant>
//#else
    #include <mpark/variant.hpp>
    namespace std {
        template<class... Args>
        using variant = mpark::variant<Args...>;
        using monostate = mpark::monostate;
        using mpark::get;
        using mpark::get_if;
    }
//#endif
```

**Comment by COx2** [comment]
This is error message.

```
CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/App.o Terra/App.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/tatsuya/Documents/Git_Repos/Terra
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -stdlib=libc++ -gmodules -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug/include -I/Users/tatsuya/Documents/Git_Repos/Terra/Terra -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/vst3sdk -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/variant/include -I/Users/tatsuya/Documents/Git_Repos/Terra/schema/cpp -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/portaudio/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/protobuf/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/fmt/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/rtmidi/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/src-library -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources/x86_64 -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug -DHAVE_TYPE_TRAITS=1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/lib/wx/include/osx_cocoa-unicode-static-3.1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/include/wx-3.1 -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX__ -D__WXOSX_COCOA__ -std=c++17 -Werror=return-type -D_DEBUG -std=c++17 -Werror=return-type -include /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/SharedPrecompiledHeaders/prefix-dyfcaglxiydfxqcsdqjtdemqfxwa/prefix.hpp -MMD -MT dependencies -MF /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/App.d --serialize-diagnostics /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/App.dia -c /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp -o /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/App.o
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp:1:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.hpp:6:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/Vst3Plugin.hpp:12:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:26:28: error: use of undeclared identifier 'mpark'
        To * As() { return mpark::get_if<To>(&data_); }
                           ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:26:42: error: 'To' does not refer to a value
        To * As() { return mpark::get_if<To>(&data_); }
                                         ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:25:24: note: declared here
        template<class To>
                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:29:40: error: use of undeclared identifier 'mpark'
        To const * As() const { return mpark::get_if<To>(&data_); }
                                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:29:54: error: 'To' does not refer to a value
        To const * As() const { return mpark::get_if<To>(&data_); }
                                                     ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/ProcessInfo.hpp:28:24: note: declared here
        template<class To>
                       ^
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp:1:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.hpp:7:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/project/Project.hpp:12:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/../device/AudioDeviceManager.hpp:4:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/../device/../misc/Either.hpp:24:30: error: use of undeclared identifier 'mpark'
    auto visit(F f) { return mpark::visit(f, data_); }
                             ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/../device/../misc/Either.hpp:27:36: error: use of undeclared identifier 'mpark'
    auto visit(F f) const { return mpark::visit(f, data_); }
                                   ^
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp:1:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.hpp:7:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/project/Project.hpp:17:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/./GraphProcessor.hpp:3:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/../misc/ThreadSafeRingBuffer.hpp:192:22: warning: unused variable 'cap' [-Wunused-variable]
        UInt32 const cap = GetCapacity();
                     ^
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp:1:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.hpp:7:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/project/Project.hpp:17:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/plugin/vst3/../../processor/../transport/../project/./GraphProcessor.hpp:135:13: warning: field 'downstream_' will be initialized after field 'upstream_channel_index_' [-Wreorder]
        ,   downstream_(downstream)
            ^
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/App.cpp:20:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:23:24: error: use of undeclared identifier 'mpark'
    To * As() { return mpark::get_if<To>(&data_); }
                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:23:38: error: 'To' does not refer to a value
    To * As() { return mpark::get_if<To>(&data_); }
                                     ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:22:20: note: declared here
    template<class To>
                   ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:26:36: error: use of undeclared identifier 'mpark'
    To const * As() const { return mpark::get_if<To>(&data_); }
                                   ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:26:50: error: 'To' does not refer to a value
    To const * As() const { return mpark::get_if<To>(&data_); }
                                                 ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:25:20: note: declared here
    template<class To>
                   ^

CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/AudioDeviceManager.o Terra/device/AudioDeviceManager.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/tatsuya/Documents/Git_Repos/Terra
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -stdlib=libc++ -gmodules -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug/include -I/Users/tatsuya/Documents/Git_Repos/Terra/Terra -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/vst3sdk -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/variant/include -I/Users/tatsuya/Documents/Git_Repos/Terra/schema/cpp -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/portaudio/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/protobuf/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/fmt/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/rtmidi/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/src-library -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources/x86_64 -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug -DHAVE_TYPE_TRAITS=1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/lib/wx/include/osx_cocoa-unicode-static-3.1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/include/wx-3.1 -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX__ -D__WXOSX_COCOA__ -std=c++17 -Werror=return-type -D_DEBUG -std=c++17 -Werror=return-type -include /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/SharedPrecompiledHeaders/prefix-dyfcaglxiydfxqcsdqjtdemqfxwa/prefix.hpp -MMD -MT dependencies -MF /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/AudioDeviceManager.d --serialize-diagnostics /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/AudioDeviceManager.dia -c /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/AudioDeviceManager.cpp -o /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/AudioDeviceManager.o
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/AudioDeviceManager.cpp:4:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/./AudioDeviceManager.hpp:4:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/../misc/Either.hpp:24:30: error: use of undeclared identifier 'mpark'
    auto visit(F f) { return mpark::visit(f, data_); }
                             ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/../misc/Either.hpp:27:36: error: use of undeclared identifier 'mpark'
    auto visit(F f) const { return mpark::visit(f, data_); }
                                   ^
2 errors generated.

CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/DeviceIOType.o Terra/device/DeviceIOType.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/tatsuya/Documents/Git_Repos/Terra
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -stdlib=libc++ -gmodules -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug/include -I/Users/tatsuya/Documents/Git_Repos/Terra/Terra -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/vst3sdk -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/variant/include -I/Users/tatsuya/Documents/Git_Repos/Terra/schema/cpp -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/portaudio/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/protobuf/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/fmt/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/rtmidi/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/src-library -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources/x86_64 -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug -DHAVE_TYPE_TRAITS=1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/lib/wx/include/osx_cocoa-unicode-static-3.1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/include/wx-3.1 -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX__ -D__WXOSX_COCOA__ -std=c++17 -Werror=return-type -D_DEBUG -std=c++17 -Werror=return-type -include /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/SharedPrecompiledHeaders/prefix-dyfcaglxiydfxqcsdqjtdemqfxwa/prefix.hpp -MMD -MT dependencies -MF /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/DeviceIOType.d --serialize-diagnostics /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/DeviceIOType.dia -c /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/DeviceIOType.cpp -o /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/DeviceIOType.o

CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiDeviceManager.o Terra/device/MidiDeviceManager.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/tatsuya/Documents/Git_Repos/Terra
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -stdlib=libc++ -gmodules -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug/include -I/Users/tatsuya/Documents/Git_Repos/Terra/Terra -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/vst3sdk -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/variant/include -I/Users/tatsuya/Documents/Git_Repos/Terra/schema/cpp -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/portaudio/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/protobuf/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/fmt/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/rtmidi/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/src-library -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources/x86_64 -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug -DHAVE_TYPE_TRAITS=1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/lib/wx/include/osx_cocoa-unicode-static-3.1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/include/wx-3.1 -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX__ -D__WXOSX_COCOA__ -std=c++17 -Werror=return-type -D_DEBUG -std=c++17 -Werror=return-type -include /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/SharedPrecompiledHeaders/prefix-dyfcaglxiydfxqcsdqjtdemqfxwa/prefix.hpp -MMD -MT dependencies -MF /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiDeviceManager.d --serialize-diagnostics /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiDeviceManager.dia -c /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.cpp -o /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiDeviceManager.o
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.cpp:1:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:23:24: error: use of undeclared identifier 'mpark'
    To * As() { return mpark::get_if<To>(&data_); }
                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:23:38: error: 'To' does not refer to a value
    To * As() { return mpark::get_if<To>(&data_); }
                                     ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:22:20: note: declared here
    template<class To>
                   ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:26:36: error: use of undeclared identifier 'mpark'
    To const * As() const { return mpark::get_if<To>(&data_); }
                                   ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:26:50: error: 'To' does not refer to a value
    To const * As() const { return mpark::get_if<To>(&data_); }
                                                 ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.hpp:25:20: note: declared here
    template<class To>
                   ^
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/MidiDeviceManager.cpp:6:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/device/../misc/ThreadSafeRingBuffer.hpp:192:22: warning: unused variable 'cap' [-Wunused-variable]
        UInt32 const cap = GetCapacity();
                     ^

CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiFile.o Terra/file/MidiFile.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/tatsuya/Documents/Git_Repos/Terra
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -stdlib=libc++ -gmodules -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Debug\" -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.14 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -Wno-semicolon-before-method-body -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug/include -I/Users/tatsuya/Documents/Git_Repos/Terra/Terra -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/vst3sdk -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/variant/include -I/Users/tatsuya/Documents/Git_Repos/Terra/schema/cpp -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/portaudio/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/protobuf/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/fmt/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/rtmidi/build_debug/install/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/include -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/midifile/src-library -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources/x86_64 -I/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Debug -DHAVE_TYPE_TRAITS=1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/lib/wx/include/osx_cocoa-unicode-static-3.1 -I/Users/tatsuya/Documents/Git_Repos/Terra/ext/wxWidgets/build_debug/install/include/wx-3.1 -D_FILE_OFFSET_BITS=64 -D__WXMAC__ -D__WXOSX__ -D__WXOSX_COCOA__ -std=c++17 -Werror=return-type -D_DEBUG -std=c++17 -Werror=return-type -include /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/SharedPrecompiledHeaders/prefix-dyfcaglxiydfxqcsdqjtdemqfxwa/prefix.hpp -MMD -MT dependencies -MF /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiFile.d --serialize-diagnostics /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiFile.dia -c /Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/MidiFile.cpp -o /Users/tatsuya/Documents/Git_Repos/Terra/build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/MidiFile.o
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/MidiFile.cpp:1:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/MidiFile.hpp:5:
In file included from /Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/Sequence.hpp:5:
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:26:28: error: use of undeclared identifier 'mpark'
        To * As() { return mpark::get_if<To>(&data_); }
                           ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:26:42: error: 'To' does not refer to a value
        To * As() { return mpark::get_if<To>(&data_); }
                                         ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:25:24: note: declared here
        template<class To>
                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:29:40: error: use of undeclared identifier 'mpark'
        To const * As() const { return mpark::get_if<To>(&data_); }
                                       ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:29:54: error: 'To' does not refer to a value
        To const * As() const { return mpark::get_if<To>(&data_); }
                                                     ^
/Users/tatsuya/Documents/Git_Repos/Terra/Terra/file/../project/../processor/ProcessInfo.hpp:28:24: note: declared here
        template<class To>
                       ^

** BUILD FAILED **


The following build commands failed:
        CompileC build_debug/Terra.build/Debug/Terra.build/Objects-normal/x86_64/AudioDeviceManager.o Terra/device/AudioDeviceManager.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/tatsuya/Documents/Git_Repos/Terra/gradle/build.gradle' line: 211

* What went wrong:
Execution failed for task ':build_app'.
> undesirable exit value 65. Expression: (result.exitValue() == 0)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

* Get more help at https://help.gradle.org
```

**Comment by hotwatermorning** [comment]
fixed in commit 0807b84

**Comment by COx2** [comment]
Thanks!
